psalm.xml

```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0"?>
+<psalm
+xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+xmlns="https://getpsalm.org/schema/config"
+autoloader="phpstan-bootstrap.php"
+>
+    <projectFiles>
+        <directory name="redaxo/src"/>
+        <ignoreFiles>
+            <directory name="redaxo/src/addons/be_style/vendor/" />
+            <directory name="redaxo/src/addons/debug/vendor/" />
+            <directory name="redaxo/src/addons/media_manager/tests/" />
+            <directory name="redaxo/src/addons/mediapool/tests/" />
+            <directory name="redaxo/src/addons/phpmailer/vendor/" />
+            <directory name="redaxo/src/addons/structure/tests/" />
+            <directory name="redaxo/src/addons/tests/vendor/" />
+            <directory name="redaxo/src/core/vendor/" />
+            <directory name="redaxo/src/core/tests/" />
+            <directory name="redaxo/src/**/fragments/*" />
```

Comment:
@muglug I am wondering why psalm reports this errors, even though we have this ignore pattern defined here:

```

393ERROR: InvalidScope - redaxo/src/addons/structure/plugins/content/fragments/slice_menu_action.php:4:30 - Invalid reference to $this in a non-class context
394$fragment->setVar('buttons', $this->items, false);
395
396
397ERROR: InvalidScope - redaxo/src/addons/structure/plugins/content/fragments/slice_menu_ep.php:4:30 - Invalid reference to $this in a non-class context
398$fragment->setVar('buttons', $this->items, false);
399
400
401ERROR: InvalidScope - redaxo/src/addons/structure/plugins/content/fragments/slice_menu_move.php:4:30 - Invalid reference to $this in a non-class context
402$fragment->setVar('buttons', $this->items, false);
403
404
405ERROR: InvalidScope - redaxo/src/addons/structure/plugins/history/fragments/history/layer.php:14:66 - Invalid reference to $this in a non-class context
406                        <div class="hidden rex-select-style"><?= $this->getVar('content1select'); ?></div>
407
408
409ERROR: InvalidScope - redaxo/src/addons/structure/plugins/history/fragments/history/layer.php:19:63 - Invalid reference to $this in a non-class context
410                            <div class="rex-select-style"><?= $this->getVar('content2select'); ?></div>
411
```

Reply:
The glob path matching only works for `<file` nodes I think?

Reply:
@muglug I tried with `<file` and it didn't work either. this `<directory` seem to work somehow for other errors - different from InvalidScope - because changing/removing this line will introduce a lot of errors which are now omitted as is